substrate/frame/delegated-staking/src/lib.rs

```diff
@@ -493,7 +493,7 @@ impl<T: Config> Pallet<T> {
 		// This should never fail but if it does, it indicates bad state and we abort.
 		T::Currency::transfer(who, &proxy_delegator, amount_to_transfer, Preservation::Expendable)?;
 
-		T::CoreStaking::update_payee(who, reward_account)?;
+		T::CoreStaking::update_payee(who, reward_account)?; // NOTE: Not sure why this is being used.
```

Comment:
Question for @Ank4n, Can this `update_payee` call be removed here? This will purely just migrate `RewardDestination::Controller` records to `RewardDestination::Account(ctlr)`, so with controllers being removed will this not be necessary anymore?

Reply:
update_payee is used to update the reward account as well right? Why should it be removed?

Reply:
ohh okay I think named the function incorrectly. It should be called `set_payee` since that is what it is really doing [internally](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/staking/src/pallet/impls.rs#L1782). Didn't realise there is an update_payee extrinsic that it can be confused with 🤦. Feel free to rename it to something else to avoid the confusion.

Reply:
Ah I see, thanks! I thought it was referring to the other `update_payee`, but can see the StakingInterface function now. 👍